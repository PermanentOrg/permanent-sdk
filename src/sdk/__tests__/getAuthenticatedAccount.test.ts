import path from "node:path";
import nock from "nock";
import { getAuthenticatedAccount } from "..";

describe("getAuthenticatedAccount", () => {
	it("should return an Account", async () => {
		nock("https://permanent.local")
			.get("/api/account/getAuthenticatedAccount")
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"getAuthenticatedAccount",
					"accountVo.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const account = await getAuthenticatedAccount({
			bearerToken: "12345",
			baseUrl: "https://permanent.local/api",
		});

		expect(account).toMatchSnapshot();
	});
});
