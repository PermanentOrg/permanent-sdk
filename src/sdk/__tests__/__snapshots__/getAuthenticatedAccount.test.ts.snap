// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`getAuthenticatedAccount should return an Account 1`] = `
{
  "createdAt": 2023-03-09T17:33:09.000Z,
  "id": 2,
  "isSftpDeletionEnabled": false,
  "updatedAt": 2023-03-09T17:33:13.000Z,
}
`;
